### ISSUE TITLE
Build failure with mtl-2.3

### ISSUE THREAD

**Issue by sjakobi** [pikajude/th-printf#12]
```
Building library for th-printf-0.7..
[ 1 of 11] Compiling Buildable        ( src/Buildable.hs, dist/build/Buildable.o, dist/build/Buildable.dyn_o )

src/Buildable.hs:12:1: warning: [-Wunused-imports]
    The import of ‘Data.Semigroup’ is redundant
      except perhaps to import instances from ‘Data.Semigroup’
    To import instances alone, use: import Data.Semigroup()
   |
12 | import           Data.Semigroup                 ( Semigroup(..) )
   | ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^

src/Buildable.hs:34:20: warning: [-Wstar-is-type]
    Using ‘*’ (or its Unicode variant) to mean ‘Data.Kind.Type’
    relies on the StarIsType extension, which will become
    deprecated in the future.
    Suggested fix: use ‘Type’ from ‘Data.Kind’ instead.
   |
34 |   type Output a :: *
   |                    ^
[ 2 of 11] Compiling Parser.Types     ( parser/Parser/Types.hs, dist/build/Parser/Types.o, dist/build/Parser/Types.dyn_o )
[ 3 of 11] Compiling Parser           ( parser/Parser.hs, dist/build/Parser.o, dist/build/Parser.dyn_o )

parser/Parser.hs:53:5: error:
    Variable not in scope: when :: Bool -> m0 () -> m ()
   |
53 |     when (oldVal == bad) $ do
   |     ^^^^
```

**Comment by sjakobi** [comment]
As a Hackage trustee I have revised v0.6 and v0.7:

* https://hackage.haskell.org/package/th-printf-0.6.0/revisions/
* https://hackage.haskell.org/package/th-printf-0.7/revisions/

**Comment by pikajude** [comment]
Should be fixed by #14. Thanks for the revisions.
